### ISSUE TITLE
Website (re)work - Part 1

### ISSUE THREAD

**Issue by garbas** [NixOS/nixos-homepage#1125]
tl;dr This is a meta ticket to track work done on the website.

In **Part 1** focus is to update the current website tech stack. There should not be any visual change. We are leaving that work for the future.

Main goal and the requirement is to have something that more people are comfortable maintaining and finding a helping hand is not an impossible task. That means we selected a stack that lowers the entry bar, has good documentation and keeps maintenance as low as possible.

The group working on this is: @garbas, @thilobillerbeck, @tfc

If anybody want to help please ping us on a [Marketing channel](https://matrix.to/#/#marketing:nixos.org)

The work for new website is happening in `main` branch (which is not the default branch at the moment). Once we are confident our work is ready to replace current `master` branch we will make `main` branch the default branch.

The stack selected is:
- [astro](https://astro.build/) as the static html generator 
- [tailwindcss](https://tailwindcss.com/) as the well known and used CSS framework
- [htmx](https://htmx.org/) as a simple "Javascript framework" replacement

You might disagree with the above stack, but this issue is not the place to discuss it, but rather to coordinate the work.

**Comment by flyfloh** [comment]
I think this is done
